integration-tests/maven/src/test/java/io/quarkus/maven/it/PackageIT.java

```diff
@@ -124,7 +125,13 @@ private void verifyUberJar() throws IOException {
         List<File> jars = getFilesEndingWith(targetDir, ".jar");
         assertThat(jars).hasSize(1);
         assertThat(getNumberOfFilesEndingWith(targetDir, ".original")).isEqualTo(1);
-
+        try (JarFile jarFile = new JarFile(jars.get(0))) {
```

Comment:
I think you need to use the other `JarFile` constructor to have support multi-release JARs read properly. Better yet, use the `io.smallrye.common.io.jar.JarFiles` class 😉

Reply:
Hello @gastaldi, in this specific testcase I was just interested in seeing if the `Multi-Release: true` attribtue made it into the jar file and for that test, using any variant of `JarFile` constructor and then calling the `isMultiRelease()` works fine.